src/debug.cpp

```diff
@@ -351,8 +434,19 @@ void realDebugmsg( const char *filename, const char *line, const char *funcname,
     if( capturing ) {
         captured += text;
     } else {
-        DebugLog( D_ERROR, D_MAIN ) << filename << ":" << line << " [" << funcname << "] " << text <<
-                                    std::flush;
+
+        if( !rep_folder.test( filename, line, funcname, text ) ) {
+            DebugLog( D_ERROR, D_MAIN ) << filename << ":" << line << " [" << funcname << "] " << text <<
+                                        std::flush;
+            rep_folder.set( filename, line, funcname, text );
+        } else {
+            rep_folder.increment_count();
+
+            // Yell at user to file a bug report if we get too high repetition count
+            if( rep_folder.repeat_count > repetition_folder::repetition_threshold ) {
+                debugmsg( "Excessive error repetition detected.  Please file a bug report at https://github.com/CleverRaven/Cataclysm-DDA/issues" );
```

Comment:
Calling `debugmsg` recursively here might have unintended side-effects. I suggest prepending this line to the original message instead. Might need to add a new flag to `debug_error_prompt` and use it in the message key to force display the error when the original error was ignored. This also has the benefit of showing the actual repeating error in the debug message.

Reply:
Probably the correct move. I'll modify to prepend the original message.
The excessive repetitions error doesn't serve much use going into `debug.log` but it would be useful as an error prompt. The excessive repetitions will become apparent when reading the log and an entry for that purpose doesn't add information.
An unintended consequence for calling `debugmsg` here I can see at the moment is for a degenerate case where a single error is repeated millions of times. This will generate folded repetition groups of
```
[ERROR] fold begin
error fold 10k
[ERROR] fold end
excessive repetition error
```
and do so a large number of times. It will save _some_ space in `debug.log` but it will become less readable as a result. It would be better to just output the prompt and skip directly logging it.

The simplest change would be to move this test down to prompt generation and make the test
`if( ( rep_folder.repeat_count % repetition_folder::repetition_threshold ) == 0 ) { ... }`
and generate a standard ignorable prompt or a forced prompt. If there is prompt forcing via flag would need to alter both `debug_error_prompt` and `replay_buffered_debugmsg_prompts` to handle the flag.
A forced, unignorable prompt would interrupt the player frequently with the prompt in a degenerate case and is likely to generate some kind of bug report here, or questions on discord or reddit which can then be used to further prompt a bug report.

Next commit:
1. Change repetition error into prompt generation so as to not prematurely end existing folds (primary PR goal)
2. Add in original error message to prompt for readability

For now I will not add in forced prompts. For now if the user wants to ignore the error they will be able to. Can be future work in a future PR if it isn't immediately required.